13/souhailshiv/Highest_Rated_Movie_Directors.py

```diff
@@ -0,0 +1,83 @@
+from collections import *
+import csv
+from urllib.request import urlretrieve
+from statistics import *
+import pandas as pd
+from termcolor import colored
+from math import *
+
+movie_data = "https://raw.githubusercontent.com/sundeepblue/movie_rating_prediction/master/movie_metadata.csv"
+movies_csv = 'movies.csv'
+urlretrieve(movie_data, movies_csv)
+Movie = namedtuple("Movie", "title year score")
+MIN_MOVIES = 4
+MIN_YEAR = 1960
+
+directors = defaultdict(list)
+
+
+def get_movies_by_director(data=movies_csv):
+    with open(data, encoding="utf-8") as f:
+        for line in csv.DictReader(f):
+            try:
+                director = line['director_name']
+                movie = line['movie_title'].replace('\xa0', '')
+                year = int(line['title_year'])
+                score = float(line['imdb_score'])
+            except ValueError:
+                continue
+            global m
+            m = Movie(title=movie, year=year, score=score)
+            directors[director].append(m)
+    return directors
+
+
+def avrage_score(directors):
+
+    cnt = Counter()
+    newDict = dict()
+
+    for director, movies in directors.items():
+        cnt[director] += len(movies)
+        c = dict(cnt)
+    for k, v in c.items():
+        if v >= MIN_MOVIES:
+            newDict[k] = v
+    avrege = dict()
+
+    for k in newDict.keys():
+        moviescores = list()
+        for item in directors[str(k)]:
```

Comment:
Item is not a good name, try to be more specific.

Why do you need a str cast here? Should not b be necessary